var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -198,6 +198,14 @@ def libs(self):
     # Container's Extended Allocators were not added until 1.56.0
     conflicts('+container', when='@:1.55.99')
 
+    # Boost.System till 1.76 (included) was reliying on mutex, which was not
+    # detected correctly on Darwin platform when using GCC
+    #
+    # More details here:
+    # https://github.com/STEllAR-GROUP/hpx/issues/5442#issuecomment-878889166
+    # https://github.com/STEllAR-GROUP/hpx/issues/5442#issuecomment-878913339
+    conflicts('%gcc', when='@:1.76 platform=darwin')
```

Comment:
Can you install `boost~system` with gcc on macos?
```suggestion
    conflicts('%gcc', when='@:1.76 +system platform=darwin')
```

Reply:
Yes, installing `boost ~system %gcc` works on macOS.